File: baml_language/sdk_tests/crates/typescript/function_calls/customizable/optional_args.static.ts
import { optional_args_probe } from "./baml_sdk/index.js";

// @ts-expect-error required arg0 cannot be omitted
optional_args_probe();

// @ts-expect-error defaulted args are passed through the $opts object
optional_args_probe(1, 8);

// @ts-expect-error unknown option key
optional_args_probe(1, { opt3: 1 });

// @ts-expect-error wrong required type
optional_args_probe("x");

// @ts-expect-error wrong option value type
optional_args_probe(1, { opt1: "x" });

File: baml_language/sdk_tests/crates/python_pydantic2/function_calls/customizable/optional_args_static.py
from baml_sdk import optional_args_probe


optional_args_probe()  # pyright: ignore[reportCallIssue]
optional_args_probe(1, 8)  # pyright: ignore[reportCallIssue]
optional_args_probe(1, opt3=1)  # pyright: ignore[reportCallIssue]
optional_args_probe("x")  # pyright: ignore[reportArgumentType]
optional_args_probe(1, opt1="x")  # pyright: ignore[reportArgumentType]
